pkg/pmem-csi-driver/controllerserver-node.go

```diff
@@ -140,62 +133,95 @@ func (cs *nodeControllerServer) CreateVolume(ctx context.Context, req *csi.Creat
 		return nil, status.Error(codes.InvalidArgument, "Name missing in request")
 	}
 
+	parameters, err := parseVolumeParameters(createVolumeParameters, req.GetParameters())
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "persistent volume: "+err.Error())
+	}
+
 	nodeVolumeMutex.LockKey(req.Name)
 	defer nodeVolumeMutex.UnlockKey(req.Name)
 
-	params := req.GetParameters()
-	if params != nil {
-		if val, ok := params["_id"]; ok {
-			/* use master controller provided volume uid */
-			volumeID = val
+	volumeID, size, err := cs.createVolumeInternal(ctx,
+		parameters,
+		req.Name,
+		req.GetVolumeCapabilities(),
+		req.GetCapacityRange(),
+	)
+	if err != nil {
+		// This is already a status error.
+		return nil, err
+	}
 
-			delete(params, "_id")
-		}
-	} else {
-		params = map[string]string{}
+	topology = append(topology, &csi.Topology{
+		Segments: map[string]string{
+			PmemDriverTopologyKey: cs.nodeID,
+		},
+	})
+
+	resp = &csi.CreateVolumeResponse{
+		Volume: &csi.Volume{
+			VolumeId:           volumeID,
+			CapacityBytes:      size,
+			AccessibleTopology: topology,
+		},
 	}
 
-	// Keeping volume name as part of volume parameters, this helps to
-	// persist volume name and to pass to Master via ListVolumes.
-	params["Name"] = req.Name
+	return resp, nil
+}
+
+func (cs *nodeControllerServer) createVolumeInternal(ctx context.Context,
+	parameters volumeParameters,
+	volumeName string,
+	volumeCapabilities []*csi.VolumeCapability,
+	capacity *csi.CapacityRange,
+) (volumeID string, actual int64, statusErr error) {
+	// Keep volume name as part of volume parameters for use in
+	// getVolumeByName.
+	parameters.name = &volumeName
 
 	// VolumeID is hashed from Volume Name if not provided by master controller.
 	// Hashing guarantees same ID for repeated requests.
+	volumeID = parameters.getVolumeID()
 	if volumeID == "" {
 		hasher := sha1.New()
-		hasher.Write([]byte(req.Name))
+		hasher.Write([]byte(volumeName))
 		volumeID = hex.EncodeToString(hasher.Sum(nil))
-		klog.V(4).Infof("Node CreateVolume: Create SHA1 hash from name:%s to form id:%s", req.Name, volumeID)
+		klog.V(4).Infof("create volume: create SHA1 hash from name: %q to form id: %s", volumeName, volumeID)
 	}
 
-	asked := req.GetCapacityRange().GetRequiredBytes()
-
-	if vol = cs.getVolumeByName(req.Name); vol != nil {
+	asked := capacity.GetRequiredBytes()
+	if vol := cs.getVolumeByName(volumeName); vol != nil {
 		// Check if the size of existing volume can cover the new request
-		klog.V(4).Infof("Node CreateVolume: Volume exists, name:%s id:%s size:%v", req.Name, vol.ID, vol.Size)
+		klog.V(4).Infof("create volume: volume exists, name: %q id: %s size: %v", volumeName, vol.ID, vol.Size)
```

Comment:
You think spelling out "Node" in log msg is not needed?
I added Node and Controller to many log msgs some time ago, and I find those still useful, even though logging change later made file names also visible (which were not when I added the prefixes). The prefixes help to see diff Node vs Controller msgs which in multiple cases have exact same wording.
You want to get rid of those prefixes now?
But this change creates non-uniform landscape as many other places still have those.

Reply:
I can add back "node" if needed.

The main intention was to make the capitalization simpler (there's no reason to write "node" with upper case here, same for "Volume") and make it easier to read with some extra spaces (but that's of course subjective).

I can take out the format string change.